autoscheduler/frontend/src/redux/actions.ts

```diff
@@ -114,3 +118,64 @@ export function updateCourseCard(index: number, courseCard: CourseCardOptions):
     return dispatch(updateCourseCardSync(index, courseCard));
   };
 }
+
+/**
+ * Sets what type of availability the user is placing on the calendar. At present, the
+ * user can only specify busy times during which they absolutely cannot have classes,
+ * but in the future, they may also be able to add preferred times to have classes.
+ * @param mode probably AvailabilityType.BUSY for now
+ */
+export function setAvailabilityMode(mode: AvailabilityType): AvailabilityModeAction {
+  return {
+    type: SET_AVAILABILITY_MODE,
+    mode,
+  };
+}
+
+export function addAvailability(availability: AvailabilityArgs): AddAvailabilityAction {
+  return {
+    type: ADD_AVAILABILITY,
+    availability,
+  };
+}
+
+/**
+ * Deletes the availbility matching the given availability args. Enter the start time
+ * as `time1` and the end time as `time2`.
+ * @param availability
+ */
+export function deleteAvailability(availability: AvailabilityArgs): DeleteAvailabilityAction {
+  return {
+    type: DELETE_AVAILABILITY,
+    availability,
+  };
+}
+
+/**
+ * Updates the availability matching the given availability args. Enter the unchanged
+ * time as time1 and the changed time as time2
+ * @param availability
```

Comment:
Could you add something like `Called for updating the bounds of availbility by dragging the edges` or something like that? Just like whenever it'll be called to give some context